pinning-services/index.js

```diff
@@ -81,73 +83,121 @@ const job = new cron.CronJob(period, function() {
         const dirWww = `${projDir}/${projName}/www`;
         const dirApi = `${projDir}/${projName}/api`;
 
-        // Pin WWW site to Hypercore
-        if (fs.existsSync(dirWww)) {
-          getDatSeed('dat-seed-www', projName, domain, txtHypercoreWww)
-            .then(seed => hyperPublisher.sync({
-              seed,
-              fsPath: dirWww,
-              drivePath: '/',
-              syncTime: 600000 // Allow up to 10 minutes for sync
-            }))
-            .then(({diff, url}) => {
-              // Log any changes to hyperdrive and return url
-              console.log(`WWW site for ${projName} pinned at ${url}. Changes:`);
-              console.log(diff);
-              return url;
-            })
-            .catch(function(error) {
-              console.log(error);
+        //Read publication settings from project configuration
+        let protoHttp = true;
+        let protoHttpDnsUpdate = true;
+        let protoIpfs = true;
+        let protoHypercore = true;
+        if (proj['publication']) {
+          if (proj['publication']['protocol']) {
+
+            // Check for known protocols and set flags
+            Object.keys(proj['publication']['protocol']).forEach((item) => {
+              switch (item) {
+                case 'http':
+                  protoHttp = (proj['publication']['protocol'][item]['enable'] === true) ? true : false;
+                  protoHttpDnsUpdate = (proj['publication']['protocol'][item]['dnsupdate'] === true) ? true : false;
+                  break;
+                case 'ipfs':
+                  protoIpfs = (proj['publication']['protocol'][item]['enable'] === true) ? true : false;
+                  break;
+                case 'hypercore':
+                  protoHypercore = (proj['publication']['protocol'][item]['enable'] === true) ? true : false;
+                  break;
+                default:
+                  console.log(`WARNING: Unknown protocol in publication project JSON - ${item}`);
```

Comment:
We should eventually come up with some standard to report errors. Like WARNING vs. ERROR. Let's leave this for now.

Reply:
May want to consider log levels and migrate away from console.log to our own logging function that can change the level of logging done. 

This way the production server wont be flooded with debug logging.

Reply:
Yup, eventually will implement